src/widget/wwidgetstack.cpp

```diff
@@ -85,13 +96,32 @@ void WWidgetStack::hideIndex(int index) {
     }
 }
 
+void WWidgetStack::showEvent(QShowEvent*) {
+    int index = static_cast<int>(m_currentPageControl.get());
+    setCurrentIndex(index);
+
+    // Set the page triggers to match the current index.
+    for (QMap<int, ControlObject*>::iterator it = m_triggers.begin();
+            it != m_triggers.end(); ++it) {
+        it.value()->set(it.key() == index ? 1.0 : 0.0);
```

Comment:
no, because each page may have a trigger CO (like a button for each page) that needs to be consistent with the show/hide state of the stack.  This code sets those trigger COs if they exist.  If the CO is incorrectly on, and the stack is shown, we need to set the CO to off.

This fixes the case where the user has samplers and microphones on and then switches to a stack where only one of them can be shown -- this code makes sure that only one of those COs is on, the one for the page that is shown.